### DISCUSSION TITLE
Creating IFC entity from data in .obj file

### DISCUSSION THREAD

**Discussion by tomerAlef** [IfcOpenShell/IfcOpenShell#2814] [General]
I'm trying to create an IFC project from the data in an .glb file (converted to obj) and I'm trying to figure out if there's a better way to do such a thing without iterating through all of the faces/vertices of the obj meshes 
(like in this example - https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.7.0/src/blenderbim/scripts/obj2ifc.py).

Is there a built in function in ifcopenshell that addressed this use case? 
(coulden't find any documentation regarding it)

**Comment by aothms** [comment]
The obj2ifc script creates a IfcFacetedBrep which is a valid construct in all schema versions, but not very efficient. Newer versions of the schema also support IfcTriangulatedFaceset which is more akin to a multi dimensional array you could get from numpy (so maybe requiring less explicit loops). I think it would be a valid addition to the script and I'm sure the community here is available to help you author it if you take the initiative. But I'm curious to ask, what is exactly your aim in finding "a better way"?

**Reply by tomerAlef** [reply]
Because of the simularity between obj and ifc entiry geometric representation I hoped there was some built in function that would give me the ability to just assign an obj to the entity as its geometric representation in the creation time

**Reply by tomerAlef** [reply]
Where can I find the documentation for the shape representations?

**Reply by aothms** [reply]
They aren't that similar, the IfcFacetedBrep with IfcFace is basically a constrained version of IfcAdvancedFace that supports curved edges and surfaces. IfcTriangulatedFaceset is strictly triangles, OBJ is polygonal. IfcPolygonalFaceset supports inner bounds which OBJ does not support.

Look for the subtypes of https://ifc43-docs.standards.buildingsmart.org/IFC/RELEASE/IFC4x3/HTML/lexical/IfcGeometricRepresentationItem.htm

**Reply by tomerAlef** [reply]
So what geometric representation do you think will be the best fit for the data contained in obj file?

**Reply by aothms** [reply]
IfcFacetedBrep  or IfcPolygonalFaceset I guess.

**Reply by tomerAlef** [reply]
Thanks
